File: 167.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        for(int i= 0; i < numbers.size()-1; i++){
            if(binary_search(numbers.begin()+i+1, numbers.end(), target - numbers[i])){
                return {i+1,(int)(lower_bound(numbers.begin()+i+1, numbers.end(), target - numbers[i]) - numbers.begin())+1};
            }
        }
        return {0,0};
    }
};

File: Python/167.py
import bisect
class Solution:
    def bs(self,arr, x):
        index = bisect.bisect_left(arr, x)
        if index != len(arr) and arr[index] == x:
            return index
        else:
            return -1

    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)-1):
            # print(numbers[i+1::])
            ind = self.bs(numbers[i+1::], target - numbers[i])
            # print(ind)
            if ind != -1:
                return [i+1, i+ind+2]
